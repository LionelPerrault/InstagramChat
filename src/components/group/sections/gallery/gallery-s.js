import React, { Component } from 'react'
import { FadeIn } from 'animate-components'
import Title from '../../../others/title'
import { connect } from 'react-redux'
import { getGroupPhotos } from '../../../../actions/post'
import End from '../../../others/end'
import { bottomScroll, cLoading } from '../../../../utils/utils'
import GroupPhotos from './photos'
import IsLoading from '../../../others/isLoading'
import classNames from 'classnames'

class GroupGallery extends Component {
  state = {
    loading: true,
  }

  componentDidMount = () => {
    let { dispatch, gd } = this.props
    dispatch(getGroupPhotos(gd.group_id))
  }

  componentWillReceiveProps = ({ dispatch, gd }) => {
    this.props.gd != gd && dispatch(getGroupPhotos(gd.group_id))
    this.setState({ loading: false })
  }

  componentDidUpdate = () => bottomScroll()

  render() {
    let { loading } = this.state,
      {
        gd: { name },
        photos,
      } = this.props,
      len = photos.length

    return (
      <div>
        <Title value={`${name} group's gallery`} />

        <FadeIn duration="300ms">
          <IsLoading loading={loading} />

          <div
            className={classNames(
              'pro_senapati',
              'photos_senapati',
              cLoading(loading)
            )}
          >
            <GroupPhotos />
          </div>

          {!loading && len != 0 && <End />}
        </FadeIn>
      </div>
    )
  }
}

const mapStateToProps = store => ({
  gd: store.Group.group_details,
  photos: store.Post.photos,
})

export default connect(mapStateToProps)(GroupGallery)
export { GroupGallery as PureGroupGallery }
