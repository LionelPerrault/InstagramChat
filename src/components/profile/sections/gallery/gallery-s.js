import React, { Component } from 'react'
import { FadeIn } from 'animate-components'
import Title from '../../../others/title'
import { connect } from 'react-redux'
import { getPhotos } from '../../../../actions/post'
import { bottomScroll, cLoading } from '../../../../utils/utils'
import End from '../../../others/end'
import PropTypes from 'prop-types'
import UserPhotos from './photos'
import IsLoading from '../../../others/isLoading'
import classNames from 'classnames'

class Gallery extends Component {
  state = {
    loading: true,
  }

  componentDidMount = () => {
    let { dispatch, ud } = this.props
    dispatch(getPhotos(ud.id))
  }

  componentWillReceiveProps = ({ dispatch, ud }) => {
    this.props.ud != ud ? dispatch(getPhotos(ud.id)) : null
    this.setState({ loading: false })
  }

  componentDidUpdate = () => bottomScroll()

  render() {
    let { loading } = this.state,
      { param: username, photos } = this.props,
      len = photos.length

    return (
      <div>
        <Title value={`${username}'s gallery`} />

        <FadeIn duration="300ms">
          <IsLoading loading={loading} />

          <div
            className={classNames(
              'pro_senapati',
              'photos_senapati',
              cLoading(loading)
            )}
          >
            <UserPhotos />
          </div>

          {!loading && len != 0 && <End />}
        </FadeIn>
      </div>
    )
  }
}

Gallery.propTypes = {
  param: PropTypes.string.isRequired,
}

const mapStateToProps = store => ({
  ud: store.User.user_details,
  photos: store.Post.photos,
})

export default connect(mapStateToProps)(Gallery)
export { Gallery as PureGallery }
