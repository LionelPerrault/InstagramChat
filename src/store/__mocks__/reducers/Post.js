import posts from './posts'

export default {
  feed: posts,
  posts,
  tagged: posts,
  shared: [
    {
      ...posts[0],
      share_id: 445,
      share_by_username: 'takkar',
      share_time: '1480114098767',
    },
  ],
  bookmarks: [posts[0]],
  photos: [
    {
      post_id: 43,
      user: 18,
      username: 'ragnar',
      firstname: 'Ragnar',
      surname: 'Lothbrok',
      post_time: '1480114098767',
      imgsrc: 'instagram_1516528062094.jpg',
      filter: 'filter-normal',
    },
  ],
  viewPost: {
    ...posts[0],
    comments: [
      {
        comment_id: 73,
        type: 'text',
        text: 'https://regexr.com/?37i6s fffffm',
        commentSrc: '',
        comment_by: 7,
        comment_by_username: 'ghalib',
        post_id: 89,
        comment_time: '1480114098767',
      },
      {
        comment_id: 69,
        type: 'image',
        text: '',
        commentSrc: 'instagram_comment_1518972851259.jpg',
        comment_by: 8,
        comment_by_username: 'coldplay',
        post_id: 89,
        comment_time: '1480114098767',
      },
      {
        comment_id: 104,
        type: 'sticker',
        text: '',
        commentSrc: 'instagram_comment_1527447929485.jpg',
        comment_by: 24,
        comment_by_username: 'takkar',
        post_id: 43,
        comment_time: '1480114098767',
      },
    ],
  },
  likes: [
    {
      like_id: 42,
      like_by: 18,
      username: 'ragnar',
      firstname: 'Ragnar',
      surname: 'Lothbrok',
      post_id: 43,
      like_time: '1480114098767',
      isFollowing: false,
    },
  ],
  tags: [
    {
      post_tag_id: 1,
      post_id: 89,
      user: 16,
      username: 'zayn',
      firstname: 'Zayn',
      surname: 'Malik',
      isFollowing: false,
    },
  ],
  isPostMine: false,
  usersToShare: [
    {
      follow_id: 302,
      follow_to: 16,
      username: 'zayn',
      firstname: 'Zayn',
      surname: 'Malik',
      didIShare: false,
    },
    {
      follow_id: 289,
      follow_to: 10,
      username: 'noddy',
      firstname: 'your',
      surname: 'noddy',
      didIShare: false,
    },
  ],
  sharers: [
    {
      share_id: 103,
      share_by: 7,
      share_by_username: 'ghalib',
      share_by_firstname: 'Mirza',
      share_by_surname: 'Ghalib',
      share_to: 28,
      share_to_username: 'selena',
      share_time: '1480114098767',
      post_id: 43,
      isFollowing: false,
    },
  ],
  postIt: {
    fileInput: '', // file input value
    fileChanged: false, // for checking file has changed
    targetFile: '', // file
    previewImg: '/images/location.jpg', // image which will be previewd
    desc: '', // textarea value
    filter: 'filter-normal',
    fetchingLocation: false,
    location: 'A-301, 90 Feet Road, Dharavi, Mumbai, Maharashtra 400017, India',
    addTag: false,
    tags: [
      {
        user: 7,
        username: 'ghalib',
      },
    ],
    showOverlay: false,
    type: 'user',
    group: 0,
  },
}
