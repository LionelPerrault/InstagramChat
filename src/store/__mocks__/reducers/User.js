export default {
  session: {
    id: 24,
    username: 'takkar',
  },
  user_details: {
    id: 24,
    username: 'takkar',
    firstname: 'iam_',
    surname: 'takkar',
    email: 'takkar@gmail.com',
    bio: '',
    account_type: 'public',
    email_verified: 'yes',
    joined: '1480114098767',
    twitter: '',
    facebook: '',
    github: 'https://github.com/yTakkar/',
    instagram: 'https://www.instagram.com/_faiyaz_shaikh',
    phone: '',
    website: '',
    isOnline: false,
    lastOnline: '',
  },
  tags: [
    {
      tag: 'traveler',
      user: 24,
    },
    {
      tag: 'nature-lover',
      user: 24,
    },
  ],
  mutualUsers: [
    {
      user: 11,
      username: 'nobita',
      firstname: 'nobita',
      surname: 'nobi',
      follow_id: 261,
      mutualUsersCount: 0,
    },
  ],
}
