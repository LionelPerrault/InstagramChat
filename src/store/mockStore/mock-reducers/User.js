export default {
  session: {
    id: 24,
    username: 'takkar',
  },
  user_details: {
    id: 7,
    username: 'ghalib',
    firstname: 'Mirza',
    surname: 'Ghalib',
    email: 'ghalib@gmail.com',
    bio: '',
    account_type: 'private',
    email_verified: 'yes',
    joined: '1514718748562',
    twitter: '',
    facebook: '',
    github: '',
    instagram: '',
    phone: '',
    website: '',
    isOnline: false,
    lastOnline: '',
  }
}
