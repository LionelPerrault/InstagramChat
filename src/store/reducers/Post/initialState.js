import PostItInitialState from './postit-initial-state'

export default {
  posts: [],
  bookmarks: [],
  tagged: [],
  shared: [],
  photos: [],
  feed: [],
  viewPost: {},
  likes: [],
  tags: [],
  posted: false,
  isPostMine: false,
  usersToShare: [],
  sharers: [],
  postIt: PostItInitialState,
}
