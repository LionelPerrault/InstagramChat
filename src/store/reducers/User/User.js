/* eslint indent:0 */
/* eslint no-unreachable:0 */

import InitialState from './intialState'
import * as methods from './methods'

export default (state = InitialState, action) => {
  let py = action.payload

  switch (action.type) {
    case 'GET_USER_DETAILS':
      return {
        ...state,
        user_details: py.details,
        tags: py.tags,
      }
      break

    case 'ADD_TAG':
      return { ...state, tags: methods.addTag(state.tags, py) }
      break

    case 'DELETE_TAG':
      return { ...state, tags: methods.deleteTag(state.tags, py) }
      break

    case 'GET_MUTUAL_USERS':
      return { ...state, mutualUsers: py }
      break
  }

  return state
}
