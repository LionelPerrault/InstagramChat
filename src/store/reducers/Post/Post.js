/* eslint indent:0 */
/* eslint no-unreachable:0 */

import initialState from './initialState'
import * as methods from './methods'

export default (state = initialState, action) => {
  let py = action.payload

  switch (action.type) {
    case 'GET_USER_POSTS':
      return { ...state, posts: py }
      break

    case 'GET_BOOKMARKED_POSTS':
      return { ...state, bookmarks: py }
      break

    case 'GET_TAGGED_POSTS':
      return { ...state, tagged: py }
      break

    case 'GET_SHARED_POSTS':
      return { ...state, shared: py }
      break

    case 'GET_PHOTOS':
      return { ...state, photos: py }
      break

    case 'GET_FEED':
      return { ...state, feed: py }
      break

    case 'GET_GROUP_POSTS':
      return { ...state, posts: py }
      break

    case 'GET_GROUP_PHOTOS':
      return { ...state, photos: py }
      break

    case 'GET_POST':
      return { ...state, viewPost: py }
      break

    case 'ADD_USER_POST':
      return {
        ...state,
        feed: methods.addPost(state.feed, py),
      }
      break

    case 'ADD_GROUP_POST':
      return {
        ...state,
        posts: methods.addPost(state.posts, py),
      }
      break

    case 'EDIT_POST':
      return {
        ...state,
        posts: methods.editPost(state.posts, py),
      }
      break

    case 'DELETE_POST':
      return {
        ...state,
        posts: methods.deletePost(state.posts, py),
        feed: methods.deletePost(state.feed, py),
      }
      break

    case 'GET_POST_LIKES':
      return {
        ...state,
        likes: py.likes,
        isPostMine: py.isPostMine,
      }
      break

    case 'REMOVE_LIKE':
      return {
        ...state,
        likes: methods.removeLike(state.likes, py),
      }
      break

    case 'GET_POST_TAGS':
      return {
        ...state,
        tags: py.tags,
        isPostMine: py.isPostMine,
      }
      break

    case 'UNTAG':
      return {
        ...state,
        tags: methods.untag(state.tags, py),
      }
      break

    case 'GET_USERS_TO_SHARE':
      return { ...state, usersToShare: py }
      break

    case 'GET_POST_SHARERS':
      return { ...state, sharers: py }
      break

    case 'UNBOOKMARK':
      return {
        ...state,
        bookmarks: methods.unbookmark(state.bookmarks, py),
      }
      break

    case 'REMOVE_SHARE':
      return {
        ...state,
        sharers: methods.removeShare(state.sharers, py),
      }
      break

    case 'COMMENT':
      return {
        ...state,
        viewPost: methods.comment(state.viewPost, py),
      }
      break

    case 'DELETE_COMMENT':
      return {
        ...state,
        viewPost: methods.deleteComment(state.viewPost, py),
      }
      break

    case 'EDIT_COMMENT':
      return {
        ...state,
        viewPost: methods.editComment(state.viewPost, py),
      }
      break

    // Post It
    case 'CHANGE_POSTIT_PROPS':
      return {
        ...state,
        postIt: methods.changePostIt(state.postIt, py),
      }

    case 'RESET_POSTIT':
      return {
        ...state,
        postIt: methods.resetPostItProperties(),
      }
  }

  return state
}
