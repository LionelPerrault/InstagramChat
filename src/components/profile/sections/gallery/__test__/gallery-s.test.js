import React from 'react'
import { PureGallery } from '../gallery-s'
import { shallow } from 'enzyme'
import ShallowRenderer from 'react-test-renderer/shallow'
import Post from '../../../../../store/mockStore/mock-reducers/Post'
import User from '../../../../../store/mockStore/mock-reducers/User'

describe('Gallery Component', () => {
  const props = {
    param: 'takkar',
    photos: Post.photos,
    ud: User.user_details
  }

  // shallow snapshot testing
  it('should match snapshot', () => {
    const renderer = new ShallowRenderer()
    const result = renderer.render(
      <PureGallery
        {...props}
      />
    )
    expect(result).toMatchSnapshot()
  })

  it('should hide spinner when loading=false', () => {
    const wrapper = shallow(
      <PureGallery
        {...props}
        dispatch={jest.fn()}
      />
    )
    wrapper.setState({ loading: false })
    expect(
      wrapper.find('IsLoading').prop('loading')
    ).toEqual(false)
  })

})
