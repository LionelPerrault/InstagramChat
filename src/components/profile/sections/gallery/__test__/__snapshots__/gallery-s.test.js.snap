// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Component should match snapshot 1`] = `
<div>
  <Connect(Title)
    value="takkar's gallery"
  />
  <FadeIn
    as="div"
    backfaceVisible="visible"
    delay="0s"
    direction="normal"
    duration="300ms"
    fillMode="none"
    iterations="1"
    playState="running"
    timingFunction="ease"
  >
    <IsLoading
      loading={true}
      when="component"
    />
    <div
      className="pro_senapati photos_senapati cLoading"
    >
      <Connect(UserPhotos) />
    </div>
  </FadeIn>
</div>
`;
